apps/dashboard/src/components/workflow-editor/base-node.tsx

```diff
@@ -143,12 +145,23 @@ const nodeVariants = cva(
   }
 );
 
-export interface BaseNodeProps extends React.HTMLAttributes<HTMLDivElement>, VariantProps<typeof nodeVariants> {}
+export interface BaseNodeProps extends React.HTMLAttributes<HTMLDivElement>, VariantProps<typeof nodeVariants> {
+  pill?: ReactNode;
+  onPillClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
+}
 
 export const Node = (props: BaseNodeProps) => {
-  const { children, variant, className, ...rest } = props;
+  const { children, variant, className, pill, onPillClick, ...rest } = props;
   return (
     <div className={nodeVariants({ variant, className })} {...rest}>
+      {pill && (
+        <div
+          className="border-neutral-alpha-200 text-foreground-600 absolute left-0 top-0 flex -translate-y-full items-center gap-1 rounded-t-lg border border-b-0 bg-neutral-50 px-1.5 py-0.5 text-xs font-medium"
+          onClick={onPillClick}
+        >
+          {pill}
+        </div>
```

Comment:
this part was taken from the trigger node as we reuse the same pill component for the conditions count